src/middleware/rewriteUrls.coffee

```diff
@@ -0,0 +1,104 @@
+url = require 'url'
+winston = require 'winston'
+Q = require 'q'
+
+Channel = (require '../../lib/model/channels').Channel
+config = require '../config/config'
+routerConf = config.get 'router'
+utils = require '../../lib/utils'
+router = require '../../lib/middleware/router'
+
+invertPathTransform = (pathTransform) ->
+  return pathTransform.replace /s\/(.*?)\/(.*?)($|\/(.*)$)/, 's/$2/$1/$4'
+
+exports.fetchRewriteConfig = (channel, authType, callback) ->
+  # set rewrite config from current channel
+  rwConfig = []
+  if channel.rewriteUrlsConfig?
+    rwConfig = channel.rewriteUrlsConfig
+
+  # add in default rewrite rules for hosts we proxy
+  utils.getAllChannels (err, channels) ->
+    if err?
+      return callback err
+
+    for channel in channels
```

Comment:
Actually, sorry, I think its right. So I only add the static config on line 18 for the current channel, then iterate through all channel and add in the 'virtual' config for hosts we proxy.